### PR TITLE
New basic public dashboard

### LINKED ISSUE #2007: New basic public dashboard
To look & feel like this:

screen shot 2015-01-30 at 19 14 53

the scope of this issue is mainly to setup a base for the layout and styles for the new public pages, applied on the public profile.

#### scope:
  - [x] feature flag (new_public_dashboard)
  - [x] base layout
    - [x] extract header bar from new-dashboard, partial?
    - [x] header map placeholder
    - [x] subheader user info
  - [x] render maps (read only, no pagination)
    - [x] map card list, 1st page (without pagination)
    - [x] tags
    - [x] set URL for item
    - [x] show change datetime diff
    - [x] zero-case
    - [x] consolidate CSS styles for map card
  - [x] static footer
  - [x] sticky footer if content don't fill window?
  - [x] add feature flag for staging/production

### PR DESCRIPTION
**Pull request by viddo**
Fixes #2007 (see it for scope)

Also noteworth for the code review:
- This is a first basic version so some features are obviously missing, my goal here was primary to setup the new public layout for the new design, applied on the default public profile page.
- It lacks all interactive features (settings dropdown etc.) so no JS features
- Restricted access by feature flag
- Reused existing Rails routes, so code path is branched in controller action to render the new layout/views
- The ruby/rails code could use a cleanup, but not sure if it's worth it at this point. new ruby/rails code is simply branching off so we can delete the "old" dashboard once its get deprecated and don't have to refactor everything again.
- Reusing a lot of CSS from dashboard, so some things from new-dashboard to new-common bundle, also tried DRY and clean up the existing rules.

### CODE REVIEW

**Comment by viddo** [comment]
@xavijam @javisantana @juanignaciosl @Kartones contains both some ruby/rails + frontend stuff, please review :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1378/))

**Comment by xavijam** [comment]
Subheader (aka filters in private dashboard) is not there as well, under control?

File: app/controllers/admin/pages_controller.rb [hunk 1]
```diff
@@ -152,18 +152,59 @@ def public
       end
     end
 
-    @tags             = viewed_user.tags(true, Visualization::Member::TYPE_DERIVED)
```
Comment (xavijam) [inline app/controllers/admin/pages_controller.rb]: We shouldn't remove @tags variable, we need it for "old" public dashboard.

File: app/models/user.rb [hunk 2]
```diff
@@ -1281,7 +1281,7 @@ def organization_user?
   end
 
   def belongs_to_organization?(organization)
-    organization_user? and self.organization.eql? organization
+    organization_user? and self.organization.id == organization.id
```
Comment (xavijam) [inline app/models/user.rb]: @juanignaciosl, could you check this?

File: app/controllers/admin/pages_controller.rb [hunk 3]
```diff
@@ -136,7 +136,7 @@ def sitemap
 
   def public
     username = CartoDB.extract_subdomain(request)
-    viewed_user = User.where(username: username.strip.downcase).first
+    viewed_user = User.find(username: username.strip.downcase)
```
Comment (juanignaciosl) [inline app/controllers/admin/pages_controller.rb]: LGTM, but [Sequel doc encourages using first instead of find (although he gives no explanation)](http://sequel.jeremyevans.net/rdoc/classes/Sequel/Model/ClassMethods.html#method-i-find). I'm not asking you to change, but I'd like to know whether there's something I should know about this change.

**Comment by xavijam** [comment]
Apart from what I've said, :+1:, new public pages on its way!

File: app/controllers/admin/pages_controller.rb (hunk 3 above)
Comment (viddo) [inline app/controllers/admin/pages_controller.rb]: Fair enough, my bad. I did change these as part of debugging the comparison issue above and simply left them as they were when I was done (since those worked fine). Since it's official recommendation I revert these changes to be consistent with existing code since it's easy to do anyway.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1380/))

File: app/models/user.rb (hunk 2 above)
Comment (viddo) [inline app/models/user.rb]: This was related to the bug I was seeing in development environment:
1. first request works fine
2. subsequent requests returns a 404, because `user.organization` do not equal the fetched organization model (due to the objects' hashes diff).

Considering this only happens in development environment is my guess that it's somehow related to Rails, but have not been able to pinpoint the internal workings to verify this thesis. Comparing the ids is probably more preferable anyway so leave it at that, but we should probably keep an eye out for this kind of odd behaviour.

The `belongs_to_organization?` method is only used for this use-case so no other code is affected by this change.

**Comment by viddo** [comment]
subheaders/filters was not included in the scope of this PR, will do it in a later issue

File: app/models/user.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/models/user.rb]: I think you can do `organization_user? and self.organization_id == organization_id` (underscores). Please check, but it would not involve fetching organization object.

File: app/controllers/admin/pages_controller.rb [hunk 4]
```diff
@@ -152,18 +152,59 @@ def public
...
+    })
+
+    @visualizations = []
+    visualizations.each do |vis|
+      @visualizations.push({
+        title:        vis.name,
+        description:  vis.description_clean,
+        id:           vis.id,
+        tags:         vis.tags,
+        url_options:  (vis.url_options.present? ? vis.url_options : Visualization::Member::DEFAULT_URL_OPTIONS),
```
Comment (juanignaciosl) [inline app/controllers/admin/pages_controller.rb]: Is this necessary? Member already has DEFAULT_URL_OPTIONS as default

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -212,4 +212,23 @@ def v2_vizjson_url(visualization)
...
+        concat ','
+      end
+    end
+    
+    if tags.size > visibleCount
+      concat "and #{tags.size - visibleCount} more"
+    end
+  end
+  
+  def name_or_username(user)
```
Comment (juanignaciosl) [inline app/helpers/application_helper.rb]: This'd be useful in the model

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1384/))

File: app/controllers/admin/pages_controller.rb (hunk 4 above)
Comment (viddo) [inline app/controllers/admin/pages_controller.rb]: Was copied form the current code, is actually not used so I'll remove it completely.

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (viddo) [inline app/helpers/application_helper.rb]: Ok, moved to user model

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1387/))

**Comment by viddo** [comment]
@juanignaciosl fixed your comments, any more concerns?
